tests/features/joinup_core/owner_deletion.feature

```diff
@@ -0,0 +1,63 @@
+@api
+Feature: As a site owner
+  In order to always have an owner assigned to a group
+  I should be able to prevent moderators from deleting all owners of a group.
+
+  Background:
+    Given users:
+      | Username       | Roles     |
+      | Group owner 1  |           |
+      | Group owner 2  |           |
+      | Group member   |           |
+      | Site moderator | moderator |
+    And the following collection:
+      | title | An owned collection |
+      | state | validated           |
+    And the following collection user memberships:
+      | collection          | user          | roles |
+      | An owned collection | Group owner 1 | owner |
+      | An owned collection | Group owner 2 | owner |
+      | An owned collection | Group member  |       |
+    And the following solution:
+      | title | An owned solution |
+      | state | validated         |
+    And the following solution user memberships:
+      | solution          | user          | roles |
+      | An owned solution | Group owner 1 | owner |
+      | An owned solution | Group owner 2 | owner |
+      | An owned solution | Group member  |       |
+
+  Scenario Outline: A privileged user cannot remove all owners of a group.
+    Given I am logged in as "Site moderator"
+    And I go to the homepage of the "<title>" <type>
+    And I click "Members"
+    And I select the "Group owner 1" row
+    And I select the "Group owner 2" row
+
+    # Try to delete all owners at once.
+    And I select "Delete the selected membership(s)" from "Action"
+    When I press "Apply to selected items"
+    Then I should see the error message "You cannot delete all owners of the <type>."
+
+    # Deleting owners when at least one remains is still possible.
+    Given I deselect the "Group owner 1" row
+    When I press "Apply to selected items"
+    Then I should see "Delete the selected membership(s) was applied to 1 item."
+
+    # Action is properly interrupted even if other memberships are about to be deleted.
+    Given I select the "Group owner 1" row
+    And I select the "Group member" row
+    And I select "Delete the selected membership(s)" from "Action"
+    When I press "Apply to selected items"
+    Then I should see the error message "You cannot delete all owners of the <type>."
```

Comment:
Can we improve this message in some way? In the standard use case of Joinup it is only possible to have 1 owner. We only have a very small number of collections with multiple owners, and this is only because they were migrated from the old platform like this.

So this means that in the 80% use case this message will be shown when the 1 single owner is deleted, and in that sense it is strange to see "You cannot delete all owners".

Reply:
Yeah, but do you have any alternative? We could have something like `You cannot proceed as that would leave the group orphaned` but that kinda involves terms that the user might be unaware of.